usr/src/uts/armv8/os/fakebop.c

```diff
@@ -479,107 +909,55 @@ read_bootenvrc(void)
 	v_len = do_bsys_getproplen(NULL, "console");
 	if (v_len > 0) {
 		(void) do_bsys_getprop(NULL, "console", consoledev);
+		if (strcmp(consoledev, "graphics") == 0) {
+			bsetprops("console", "text");
+			v_len = strlen("text");
+			bcopy("text", consoledev, v_len);
+		}
 	} else {
 		v_len = 0;
 	}
 	consoledev[v_len] = 0;
-}
-static void
-bsetprop(char *name, int nlen, void *value, int vlen)
-{
-	pnode_t chosen = prom_chosennode();
-	prom_setprop(chosen, name, (const caddr_t)value, vlen);
-}
 
-static void
-bsetprops(char *name, char *value)
-{
-	bsetprop(name, strlen(name), value, strlen(value) + 1);
-}
-
-static void
-bsetprop64(char *name, uint64_t value)
-{
-	bsetprop(name, strlen(name), (void *)&value, sizeof (value));
-}
-
-static void
-bsetpropsi(char *name, int value)
-{
-	char prop_val[32];
-
-	(void) snprintf(prop_val, sizeof (prop_val), "%d", value);
-	bsetprops(name, prop_val);
-}
-
-/*
- * to find the size of the buffer to allocate
- */
-/*ARGSUSED*/
-int
-do_bsys_getproplen(bootops_t *bop, const char *name)
-{
-	pnode_t chosen = prom_chosennode();
-	return prom_getproplen(chosen, name);
-}
+#if XXXARM
+	bcons_post_bootenvrc(inputdev, outputdev, consoledev);
+#endif
 
-int
-do_bsys_getprop(bootops_t *bop, const char *name, void *value)
-{
-	pnode_t chosen = prom_chosennode();
-	prom_getprop(chosen, name, (caddr_t)value);
-	return 0;
+#if XXXARM
```

Comment:
I don't know why we can't do this part yet?

Reply:
See the previous comment - I'm trying to not diverge too far from the i86pc structure, so when we get `boot_console.c` I'll enable this.